src/components/RunGraph.vue

```diff
@@ -4,25 +4,22 @@
 
 <script lang="ts" setup>
   import { onBeforeUnmount, onMounted, ref } from 'vue'
-  import { RunGraphConfig } from '@/models/RunGraph'
-  import { start, stop } from '@/objects'
-  import { RunGraphDomain } from '@/objects/domain'
+  import { RunGraphProps } from '@/models/RunGraph'
+  import { ScaleXDomain, start, stop } from '@/objects'
   import { emitter } from '@/objects/events'
-  // import { WorkerMessage, worker } from '@/workers/runGraph'
 
-  const props = defineProps<{
-    config: RunGraphConfig,
-    domain?: RunGraphDomain,
-  }>()
+  // using the props object as a whole
+  // eslint-disable-next-line vue/no-unused-properties
+  const props = defineProps<RunGraphProps>()
 
   const emit = defineEmits<{
-    (event: 'update:domain', value: RunGraphDomain): void,
+    (event: 'update:viewport', value: ScaleXDomain): void,
```

Comment:
I'm surprised to see `ScaleXDomain` as the emitted type here; the emitter is pass-through emitting the X-scale range just below this, which I think makes more sense given that the viewport is no longer tied to the domain.

This also doesn't take into account the Y scale range, which I know won't be continuous but is probably important given the layout can extend beyond the viewport in both directions

Reply:
`ScaleXDomain` is just a date tuple. Happens to be the same type as the viewport visible dates. Should I make a separate type? 

Hadn't considered the y axis since our current implementation only syncs the x axis. Probably could expose that, however might get weird because the viewport is a specific size (set by the canvas element). So you have to maintain that aspect ratio. So if you changed the dates you'd have to change the y axes to match. Which would get weird IMO.

Reply:
Yeah I guess I'd expect the viewport v-model to reflect the entirety of the viewport and not one dimension of it, though I recognize it feels like the language should be the same; I don't think there's an immediate need to address this just thinking through camera controls later down the line, they could get pretty wonky or be very limited in this implementation

Reply:
yeah makes sense. Just not something we need right now I don't think.